tests/rbac_test.go

```diff
@@ -0,0 +1,212 @@
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ktests "kubevirt.io/kubevirt-ansible/tests"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("RBAC", func() {
+	flag.Parse()
+
+	var vmi ktests.VMManifest
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	BeforeEach(func() {
+		By("Login with the user test_admin")
+		args := []string{"login", "-u", "test_admin", "-p", "123456"}
+		_, err := tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		tests.BeforeTestCleanup()
+		vmi.Name = "vm-cirros"
+		vmi.Manifest = "tests/manifests/vm-cirros.yaml"
+		ktests.CreateUser()
+
+		By("Create a VM with yaml")
+		args = []string{"create", "-f", vmi.Manifest, "-n", tests.NamespaceTestAlternative}
+		_, err = tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Check VMI should create")
+		args = []string{"start", vmi.Name, "-n", tests.NamespaceTestAlternative}
+		_, err = tests.RunCommand("virtctl", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Check the vmi is running")
+		args = []string{"get", "vmi", vmi.Name, "--template", "{{.status.phase}}", "-n", tests.NamespaceTestAlternative}
+		Eventually(func() string {
+			output, err := tests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+			return output
+		}, time.Minute*10).Should(Equal("Running"))
+	})
+
+	AfterEach(func() {
+		By("Login with the user test_admin")
+		args := []string{"login", "-u", "test_admin", "-p", "123456"}
+		_, err := tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		ktests.DeleteUser()
+
+		By("Delete identity of user kubevirt-test-user")
+		args = []string{"delete", "identity", "allow_all_auth:kubevirt-test-user"}
+		_, err = tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	DescribeTable("RBAC Authorization", func(role, namespace string, con bool) {
+		By(fmt.Sprintf("Add %s to user kubevirt-test-user", role))
+		args := []string{"adm", "policy", "add-role-to-user", role, ktests.UsernameTestUser, "-n", namespace}
+		_, err := tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Login with the user kubevirt-test-user")
+		args = []string{"login", "-u", ktests.UsernameTestUser, "-p", "123456"}
+		_, err = tests.RunCommand("oc", args...)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Create a VM with yaml")
+		args = []string{"create", "-f", vmi.Manifest, "-n", namespace}
+		_, err = tests.RunCommand("oc", args...)
+		if con == true {
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
@adityaramteke The CI seems passing, but I see some issues with the details
```
    with edit permission
    [90m/root/go/src/kubevirt.io/kubevirt-ansible/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46[0m
[90m------------------------------[0m
level=error timestamp=2018-09-28T14:21:22.193610Z pos=utils.go:1461 component= output="Error from server (Forbidden): error when creating \"tests/manifests/vm-cirros.yaml\": virtualmachines.kubevirt.io is forbidden: User \"kubevirt-test-user\" cannot create virtualmachines.kubevirt.io in the namespace \"kubevirt-test-alternative\": User \"kubevirt-test-user\" cannot create virtualmachines.kubevirt.io in project \"kubevirt-test-alternative\"\n" reason="exit status 1" msg="oc command failed: /usr/bin/oc create -f tests/manifests/vm-cirros.yaml -n kubevirt-test-alternative,"
level=error timestamp=2018-09-28T14:21:22.238658Z pos=utils.go:1461 component= output="Error: VirtualMachineInstance 'vm-cirros' is already running\n" reason="exit status 1" msg="virtctl command failed: /usr/bin/virtctl start vm-cirros -n kubevirt-test-alternative,"
level=error timestamp=2018-09-28T14:21:22.314277Z pos=utils.go:1461 component= output="Error updating VirtualMachine: virtualmachines.kubevirt.io \"vm-cirros\" is forbidden: User \"kubevirt-test-user\" cannot patch virtualmachines.kubevirt.io in the namespace \"kubevirt-test-alternative\": User \"kubevirt-test-user\" cannot \"patch\" \"virtualmachines.kubevirt.io\" with name \"vm-cirros\" in project \"kubevirt-test-alternative\"\n" reason="exit status 1" msg="virtctl command failed: /usr/bin/virtctl stop vm-cirros -n kubevirt-test-alternative,"
level=error timestamp=2018-09-28T14:21:22.444743Z pos=utils.go:1461 component= output="Error from server (Forbidden): error when deleting \"tests/manifests/vm-cirros.yaml\": virtualmachines.kubevirt.io \"vm-cirros\" is forbidden: User \"kubevirt-test-user\" cannot delete virtualmachines.kubevirt.io in the namespace \"kubevirt-test-alternative\": User \"kubevirt-test-user\" cannot delete virtualmachines.kubevirt.io in project \"kubevirt-test-alternative\"\n" reason="exit status 1" msg="oc command failed: /usr/bin/oc delete -f tests/manifests/vm-cirros.yaml -n kubevirt-test-alternative,"
[32mâ€¢ [SLOW TEST:400.777 seconds][0m```
Should `edit` permission be allowed to create vm? the error seems conflicting
About starting the VM, it seems VM from `admin` was not cleaned up, need to add that

Reply:
@vatsalparekh Yes, the `edit` permission should able to create VM. In this case, the error should occur because of the `view` permission available with the user.

Reply:
@adityaramteke but the error seems under edit permission, view permission had different logs

Reply:
@vatsalparekh The logs that you are showing are panicked errors from `view` permission test. Please have a look again or see Shiyang's test result for your clarification that errors always get panicked before test execution. 
@shiywang @gouyang please suggest on this if I'm wrong.

Reply:
@vatsalparekh yes, the logs are from `view` testing and it's not coming from ginko, it's error message from `oc`.